### ISSUE TITLE
Question about the pre-trained weights

### ISSUE THREAD

**Issue by LZL-CS** [jike5/LATITUDE#2]
Hi @Tao-11-chen, 

Thanks for your excellent work!
Can you share your trained weights for the pose_regressin model and pose_optimation model so that I can evaluate the results?

**Comment by Tao-11-chen** [comment]
Sure, the pre_trained weights will be pulished soon. I,ll test this method on public datasets like Cambridge and 7scenes too.

**Comment by LZL-CS** [comment]
@Tao-11-chen Thanks for your reply. I trained all the models based on the READ.me file. But I have some questions:

1. When I was training the module "Pose Regressor", I visualized the loss by Tensorboard and got the final best results about rotation and translation. The loss curve is quite volatile. Do you think there is something wrong with this result?
<img width="716" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/825d5f81-a0be-4fad-aa30-cb0e63ea868a">
<img width="1039" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/a69df66c-0def-492c-b92c-fdae2e59f1ec">

<img width="476" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/ac7af2a1-3273-4ef2-a394-a3bd04b82601">


2. In the module "Pose Optimization", it seems that your code doesn't use the init_poses trained by the "Pose Regressor" module. Actually, you just perturb ground truth poses, then use it as init_poses as input to trained mega_nerf, which is the same way as iNeRF. Thus I am confused as to why it is different from your paper's architecture.
<img width="979" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/7ee9a747-d172-4d8c-badc-6ce27bb4259f">

Maybe I missing something important, I appreciate any help you can provide!

**Comment by jike5** [comment]
@LZL-CS Hello, I can answer your question two.
In the paper, there are two experiments related to  pose optimization, one in Experiment C section and one in Experiment D section. The code we provide is by default performed with the Experiment D settings, which is 'just perturb ground truth poses', but of course you can set this parameter [pose_regressor_input](https://github.com/jike5/LATITUDE/blob/70d91af9a3ddc112fb523a0dc559c57e43abcc03/pose_optimization/utils.py#L25) to run with the PlaceRecognition as initial value

**Comment by LZL-CS** [comment]
> @LZL-CS Hello, I can answer your question two. In the paper, there are two experiments related to pose optimization, one in Experiment C section and one in Experiment D section. The code we provide is by default performed with the Experiment D settings, which is 'just perturb ground truth poses', but of course you can set this parameter [pose_regressor_input](https://github.com/jike5/LATITUDE/blob/70d91af9a3ddc112fb523a0dc559c57e43abcc03/pose_optimization/utils.py#L25) to run with the PlaceRecognition as initial value

@jike5 Hi, thanks for your reply, I will check it now.

**Comment by Tao-11-chen** [comment]
@LZL-CS The images generated by nerf for data augmentation are random for pose regressor, which makes the loss volatile. However, your result is a bit strange, i'll check the code later.

**Comment by LZL-CS** [comment]
> @LZL-CS The images generated by nerf for data augmentation are random for pose regressor, which makes the loss volatile. However, your result is a bit strange, i'll check the code later.

@Tao-11-chen Hi, thanks for your reply. 

1. Firstly, I visualize the rendered results from the trained mega_nerf model, and it looks well:
<img width="1032" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/89446282-1de8-4982-841a-5e6f6b0f381a">

<img width="742" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/75186989-b2e2-4293-8137-1a2d44133205">


2. Then, in **Pose Optimization**, I set the `--pose_regressor_input` option to use the **Pose Regressor** module's output as the initial value. I re-train the model, but the results seemed so weird as before:
<img width="1077" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/0367b881-0fd3-4dcd-96b3-10909ae22b09">

<img width="1029" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/28560f05-5ed9-49ef-91a3-a9c4c787b031">

<img width="247" alt="image" src="https://github.com/jike5/LATITUDE/assets/54640259/96dcda26-6e4d-4f4a-9656-b7780ae4b23c">


Maybe I missing some key points, and  I appreciate any help you can provide to resolve this issue!

**Comment by Tao-11-chen** [comment]
@LZL-CS Sorry, it's a bug in our code, i'll update the code these days.

**Comment by LZL-CS** [comment]
> @LZL-CS Sorry, it's a bug in our code, i'll update the code these days.

Hi @Tao-11-chen, may I know when you can update this code? Thanks!

**Comment by Tao-11-chen** [comment]
@LZL-CS 
We sincerely apologize that our method is not as robust as initially thought. This was something we overlooked initially, it requires the regressor to provide more stable initial poses. However, after refactoring the entire code, the performance of the regressor has declined. It appears to be a parameter issue, but the random generation of the data augmentation generated by NERF within the regressor makes it hard to reproduce previous results. It might be helpful to replace the regressor with a more powerful one. Once again, we apologize for any inconvenience.
